lib/wanda/execution/evaluation.ex

```diff
@@ -0,0 +1,169 @@
+      |> Catalog.get_expectations()
+      |> Enum.map(&eval_expectation(&1, facts))
+
+    %AgentCheckResult{
+      agent_check_result
+      | expectation_evaluations: expectation_evaluations
+    }
+  end
+
+  defp eval_expectation(%Expectation{name: name, type: type, expression: expression}, facts) do
+    case Abacus.eval(expression, facts) do
+      {:ok, return_value} ->
+        %ExpectationEvaluation{name: name, type: type, return_value: return_value}
+
+      {:error, :einkey} ->
+        %ExpectationEvaluationError{
+          name: name,
+          message: "Fact is not present.",
+          type: :fact_missing_error
+        }
+
+      _ ->
+        %ExpectationEvaluationError{
+          name: name,
+          message: "Illegal expression provided, check expression syntax.",
+          type: :illegal_expression_error
+        }
+    end
+  end
+
+  defp add_expectation_evaluations(
+         %CheckResult{check_id: check_id, agents_check_results: agents_check_results} = result
+       ) do
+    expectation_results =
+      agents_check_results
+      |> Enum.flat_map(fn %AgentCheckResult{expectation_evaluations: expectation_evaluations} ->
+        expectation_evaluations
+      end)
+      |> Enum.group_by(& &1.name)
+      |> Enum.map(fn {name, expectation_evaluations} ->
+        %{type: type} = Catalog.get_expectation(check_id, name)
+
+        %ExpectationResult{
+          name: name,
+          type: type,
+          result: eval_expectation_result_or_error(type, expectation_evaluations)
+        }
+      end)
+
+    %CheckResult{result | expectation_results: expectation_results}
+  end
+
+  defp eval_expectation_result_or_error(type, expectation_evaluations) do
+    if has_error?(expectation_evaluations) do
+      false
+    else
+      eval_expectation_result(type, expectation_evaluations)
+    end
+  end
+
+  defp has_error?(expectation_evaluations) do
+    Enum.any?(expectation_evaluations, fn
+      %ExpectationEvaluationError{} -> true
+      _ -> false
+    end)
+  end
+
+  defp eval_expectation_result(:expect_same, expectation_evaluations) do
+    expectation_evaluations
+    |> Enum.uniq_by(& &1.return_value)
+    |> Kernel.length() == 1
+  end
+
+  defp eval_expectation_result(:expect, expectations_evaluations) do
+    Enum.all?(expectations_evaluations, &(&1.return_value == true))
+  end
+
+  defp aggregate_result(%CheckResult{expectation_results: expectation_results} = check_result) do
+    result =
+      if Enum.all?(expectation_results, &(&1.result == true)) do
+        :passing
+      else
+        :critical
+      end
+
+    %CheckResult{check_result | result: result}
+  end
+
+  defp aggregate_execution_result(%Result{check_results: check_results} = execution_result) do
+    result =
+      check_results
+      |> Enum.map(& &1.result)
+      |> Enum.map(&{&1, result_weight(&1)})
+      |> Enum.max_by(fn {_, weight} -> weight end)
+      |> elem(0)
+
+    %Result{execution_result | result: result}
```

Comment:
What to we store here at the end? `:passing|:warning...` or the weight? I guess the 1st one, but I don't really understand how hehe

Reply:
this will store the `:passing | :warning ...`
you can check it here: https://github.com/trento-project/wanda/pull/4/commits/b22b534147d771dc8a479d312228d030939290f8#diff-0a5bf24059f0d65a81c7c45f97bbf5c1b712a07e66e1044fe14a85db757ebbf8

always refer to the type specs because the dialyzer catches if we are using values other than the specified